kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BaseClient.java

```diff
@@ -39,10 +40,30 @@
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.function.Predicate;
 
 public abstract class BaseClient implements Client {
 
+  /**
+   * An {@link ExecutorSupplier} that provides an unlimited thread pool {@link Executor} per client.
+   */
+  public static final ExecutorSupplier DEFAULT_EXECUTOR_SUPPLIER = new ExecutorSupplier() {
+
+    @Override
+    public Executor get() {
+      return Executors.newCachedThreadPool(Utils.daemonThreadFactory(this));
```

Comment:
I'm concerned about this one.
When is the DEFAULT_EXECUTOR_SUPPLIER.get() method going to be invoked during the lifecycle of the application?
Should we just keep a single instance? Is there a use-case where we want to maintain more than one Executor?

Reply:
It is called once per client.  With the way things are currently every call to build() creates a fresh client.  If the builder is only allowed to create a single client, then it could just be a single instance.

Reply:
Shall we switch this to create lazily loaded singleton instance?

Reply:
KubernetesClientBuilder is new, so whatever we do there won't affect existing users.  However that is functionally different than many of the other Builders, including the HttpClient.Builder.  Because you can still call the with methods after build it would probably be clearest if subsequent calls to build through an illegalstateexception.

Reply:
> it would probably be clearest if subsequent calls to build through an illegalstateexception.

Yup, I think that's even better.

Reply:
Let's take that up as a follow-on task.